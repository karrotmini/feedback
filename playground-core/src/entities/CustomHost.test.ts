import { describe, expect } from 'vitest';

import {
  CustomHost,
  CustomHostID,
  HostnameProviderInfo,
} from '../entities';

describe('CustomHost', test => {
  test('should be valid after CustomHostProvisioned', () => {
    const id = CustomHostID('TEST');
    const providerInfo = new HostnameProviderInfo({
      hostname: 'test.karrotmini.app',
      healthCheckUrl: 'https://management',
      managementUrl: 'https://healthcheck',
    });

    const customHost = new CustomHost(id);
    customHost.$publishEvent({
      aggregateId: id,
      aggregateName: 'CustomHost',
      eventName: 'CustomHostProvisioned',
      eventDate: new Date('2022-06-10').getTime(),
      eventPayload: {
        providerInfo: providerInfo.toJSON(),
      },
    });

    expect(customHost.$valid).toBe(true);
    expect(customHost.$snapshot).toMatchSnapshot(
      `$snapshot version ${customHost.snapshotVersion}`,
    );
  });

  test('create without appId', () => {
    const id = CustomHostID('TEST');
    const providerInfo = new HostnameProviderInfo({
      hostname: 'test.karrotmini.app',
      healthCheckUrl: 'https://management',
      managementUrl: 'https://healthcheck',
    });

    const customHost = CustomHost.createWithProviderInfo({
      id,
      providerInfo,
    });

    expect(customHost.toJSON()).toEqual({
      id,
      providerInfo: providerInfo.toJSON(),
      connectedApp: null,
    });
  });

  test('create with appId', () => {
    const id = CustomHostID('TEST');
    const providerInfo = new HostnameProviderInfo({
      hostname: 'test.karrotmini.app',
      healthCheckUrl: 'https://management',
      managementUrl: 'https://healthcheck',
    });

    const customHost = CustomHost.createWithProviderInfo({
      id,
      providerInfo,
    });

    expect(customHost.toJSON()).toEqual({
      id,
      providerInfo: providerInfo.toJSON(),
      connectedApp: null,
    });
  });
});
