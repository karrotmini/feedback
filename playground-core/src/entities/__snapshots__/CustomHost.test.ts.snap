// Vitest Snapshot v1

exports[`CustomHost > should be valid after CustomHostProvisioned > $snapshot version 1 1`] = `
{
  "connectedApp": null,
  "createdAt": 1654819200000,
  "providerInfo": {
    "healthCheckUrl": "https://management/",
    "hostname": "test.karrotmini.app",
    "managementUrl": "https://healthcheck/",
  },
}
`;
